### PR TITLE
minor: fixed debug option bleeding into other tests

### PR DESCRIPTION
**Pull request by rnveach**
This fixes the issue seen running pitest on CS in Windows.

The 3 tests that were causing the problem for me were, in the specific order:
*    com.puppycrawl.tools.checkstyle.MainTest.testDebugOption
*    com.puppycrawl.tools.checkstyle.internal.XDocsPagesTest.testAllCheckSections
*    com.puppycrawl.tools.checkstyle.internal.XDocsPagesTest.testAllXmlExamples

Which caused the following exception in Windows, but not on Linux:
`FINE : MINION : FAIL Description [testClass=com.puppycrawl.tools.checkstyle.internal.XDocsPagesTest, name=testAllXmlExamples(com.puppycrawl.tools.checkstyle.internal.XDocsPagesTest)] -> java.lang.OutOfMemoryError: Java heap space`

Basically our junit testing the debug option is bleeding into other tests, leaving logging on the highest. `XDocsPages` test run alot of code that logs a lot of ignorable exceptions which is why it is the point of failure.

Pitest uses its own custom ordering for junits, so that coupled with the modifications it makes to the code is probably why travis or I haven't run into this before.

Full discussion on pitest: https://github.com/hcoles/pitest/issues/272

### CODE REVIEW

**Comment by rnveach** [comment]
Travis failure is just the issue with NoExceptionTest.
